drake/systems/frames/@CoordinateFrame/new.cpp

```diff
@@ -0,0 +1,20 @@
+#include "drakeMexUtil.h"
+#include "CoordinateFrame.h"
+
+using namespace std;
+
+void mexFunction( int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[] ) {
+  if (nrhs==1) {
+    destroyDrakeMexPointer<CoordinateFrame*>(prhs[0]);  // note: prhs[0] is the actual DrakeMexPointer object (not the CoordinateFrame object, as it would be in the non-static class methods)
```

Comment:
I am a bit confused here, so if we call `new(coordinate_frame_object.mex_ptr)` in MATLAB, it would destruct the mex pointer, but leave the MATLAB object unchanged? That feels a bit weird to me, ad `new` can act as both constructor or destructor, depending on the number of inputs? Would it be better to use two different functions for the constructor and cleanup separately?